### ISSUE TITLE
BR: Fix internal server error for malformed locations input

### ISSUE THREAD

**Issue by a-stacey** [data61/Mirza#354]
Fix location parser error.
The following:
```
curl -X POST "http://localhost:8200/location/add" -H  "accept: application/json;charset=utf-8" -H  "Content-Type: application/json;charset=utf-8" -d "{  \"newLocGLN\": {    \"_sglnCompanyPrefix\": \"prefix\",    \"_locationRef\": {\"_locationRefVal\" : \"test\"},    \"_sglnExt\": \"test\"  },  \"newLocCoords\": [123,456  ],  \"newLocAddress\": \"string\"}"
```
results in:
```
HTTP/1.0 500 Internal Server Error
Date: Wed, 09 Jan 2019 05:28:54 GMT
Server: Warp/3.2.22
Content-Type: text/plain; charset=utf-8

Something went wrong
```
With the following logged from the server
```
InvalidLength (XMLSnippet "prefix.test.test")
```

The original JSON that was used in swagger is:
```
{
  "newLocGLN": {
    "_sglnCompanyPrefix": "prefix",
    "_locationRef": {
"_locationRefVal" : "test"
},
    "_sglnExt": "test"
  },
  "newLocCoords": [123,456
  ],
  "newLocAddress": "string"
}
```

**Comment by a-stacey** [comment]
Here is a valid location:
```
{
  "newLocCoords":[1,2],
  "newLocAddress": "42 Wallby Way, Sydney", 
  "newLocGLN": {
    "_sglnExt":null,
    "_locationRef": {
      "_locationRefVal": "98765"
    },
  "_sglnCompanyPrefix":"5000001"
  }
}
```

It seems that it might be when the `"_sglnExt"` was not present in the `"newLocGLN"`

**Comment by paulsamways** [comment]
Ashley: I'm not convinced this is an accurate description of the issue / steps to reproduce.

**Comment by a-stacey** [comment]
Based on some testing just now it seems that there is an issue if the company prefix is not 7 charcters. The GS1Company prefix is not validated anywhere else in the BR -> underlying GS1 types implementation, but here it seems that if you use a company prefix with anything but 7 characters (doesn't seem to matter if it is not digits, letters also seem to work) that it fails with the `InvalidLength (XMLSnippet "` error and so the above JSON structure being invalid could be completely wrong (still need to test this at some point). Its also possible that there are other errors, but testing today seems to confirm at least the number of characters must be exactly 7.

**Comment by a-stacey** [comment]
Yep cool, the constraint makes sense against this behavior. Still need to fix the implementation so that it returns something more sensible then: 500 Something went wrong.

If you happen to have the specific ref to hand  in the standard that might be nice to include here to "According to the spec, the total length of GS1CompanyPrefix and LocationRefVal needs to be equal to 12".
